### ISSUE TITLE
fraction randomly instead of slice

### ISSUE THREAD

**Issue by johnnynunez** [ultralytics/ultralytics#8233]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

If you have several joined datasets, you can delete one of them with slice.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@johnnynunez hello! 👋

It sounds like you're interested in using a fraction of your datasets randomly rather than slicing them. You can achieve this by setting the `fraction` parameter in your training configuration. This parameter allows you to specify the fraction of the dataset to use for training. For example, if you want to train on 50% of your data randomly selected, you can set `fraction=0.5` in your training command or YAML file.

Here's a quick example of how to set it in your training command:

```bash
yolo detect train data=your_dataset.yaml model=yolov8n.pt epochs=100 fraction=0.5
```

This will randomly select half of your data for training. Hope this helps, and we look forward to your PR! 😊

**Comment by johnnynunez** [comment]
> @johnnynunez hello! 👋
> 
> It sounds like you're interested in using a fraction of your datasets randomly rather than slicing them. You can achieve this by setting the `fraction` parameter in your training configuration. This parameter allows you to specify the fraction of the dataset to use for training. For example, if you want to train on 50% of your data randomly selected, you can set `fraction=0.5` in your training command or YAML file.
> 
> Here's a quick example of how to set it in your training command:
> 
> ```shell
> yolo detect train data=your_dataset.yaml model=yolov8n.pt epochs=100 fraction=0.5
> ```
> 
> This will randomly select half of your data for training. Hope this helps, and we look forward to your PR! 😊

Where is randomly on your code’ I’ve seen only slice

**Comment by Burhan-Q** [comment]
@johnnynunez I believe I understand the change you're proposing. I suspect that the difference is slightly nuanced in what happens now vs what you're proposing. Your proposed change would randomize the selections for the `train`/`val` split. The existing code uses the `shuffle` argument in the `Dataloader` but I'm not certain that this would accomplish the same result, since the randomization is within the `Dataloader` which is probably specific to either `train` or `val` set (I'm not 100% certain). 

I suspect that in _most_ cases, people who care enough to want randomized `train` and `val` sampling, will probably split the data before passing in for training. Anyone else using `fraction` either doesn't care about randomizing or has a large enough dataset that splitting it this way wouldn't matter all that much. 

Out of curiosity, have you tested to measure how much process time would be added using `random.sample` to split the dataset? I suspect this would be one of the most important considerations when evaluating this change.

**Comment by Burhan-Q** [comment]
Closed by #8234
